akka-http/src/main/scala/akka/http/javadsl/server/directives/SecurityDirectives.scala

```diff
@@ -259,6 +259,10 @@ abstract class SecurityDirectives extends SchemeDirectives {
   /**
    * Creates a `Basic` [[HttpChallenge]] for the given realm.
    */
-  def challengeFor(realm: String): HttpChallenge = HttpChallenge.create("Basic", realm)
+  def basicChallengeFor(realm: String): HttpChallenge = HttpChallenge.create("Basic", realm)
```

Comment:
Do we really need these methods here?
I think we could safely remove them. 

Perhaps adding nicer methods to `HttpChallange`
 instead.